src/jestUtils.ts

```diff
@@ -0,0 +1,359 @@
+    if (previousEvent.state !== currentEvent.state) {
+      invariant(
+        hasProperty(currentEvent, 'oldState'),
+        errorMsgWithCurrentEvent(
+          'when state changes, oldState field should be present'
+        )
+      );
+      invariant(
+        currentEvent.oldState === previousEvent.state,
+        errorMsgWithBothEvents(
+          "when state changes, oldState should be the same as previous event' state"
+        )
+      );
+    }
+  }
+
+  return currentEvent;
+}
+
+interface HandlerInfo {
+  handlerType: HandlerNames;
+  handlerTag: number;
+}
+function fillMissingDefaultsFor({
+  handlerType,
+  handlerTag,
+}: HandlerInfo): (
+  event: Partial<GestureHandlerTestEvent>
+) => Omit<GestureHandlerTestEvent, 'state' | 'oldState'> {
+  return (event) => {
+    return {
+      ...handlersDefaultEvents[handlerType],
+      ...event,
+      handlerTag,
+    };
+  };
+}
+
+type EventEmitter = (
+  eventName: string,
+  args: { nativeEvent: GestureHandlerTestEvent }
+) => void;
+interface HandlerData {
+  emitEvent: EventEmitter;
+  handlerType: HandlerNames;
+  handlerTag: number;
+}
+function getHandlerData(
+  componentOrGesture: ReactTestInstance | GestureType
+): HandlerData {
+  if (isGesture(componentOrGesture)) {
+    const gesture = componentOrGesture;
+    return {
+      emitEvent: (eventName, args) => {
+        DeviceEventEmitter.emit(eventName, args.nativeEvent);
+      },
+      handlerType: gesture.handlerName as HandlerNames,
+      handlerTag: gesture.handlerTag,
+    };
+  }
+  const gestureHandlerComponent = componentOrGesture;
+  return {
+    emitEvent: (eventName, args) => {
+      fireEvent(gestureHandlerComponent, eventName, args);
+    },
+    handlerType: gestureHandlerComponent.props.handlerType as HandlerNames,
+    handlerTag: gestureHandlerComponent.props.handlerTag as number,
+  };
+}
+declare global {
+  interface Array<T> {
+    withPrevAndCurrent: <Transformed>(
+      mapFn: withPrevAndCurrentMapFn<T, Transformed>
+    ) => Transformed[];
+  }
+}
+
+function extendArrayProto(propertyName: string, f: unknown) {
+  // eslint-disable-next-line no-extend-native
+  Object.defineProperty(Array.prototype, propertyName, {
+    configurable: true,
+    value: f,
+  });
+}
+
+export function fireGestureHandler(
+  componentOrGesture: ReactTestInstance | GestureType,
+  eventList: Partial<GestureHandlerTestEvent>[]
+): void {
+  const { emitEvent, handlerType, handlerTag } = getHandlerData(
+    componentOrGesture
+  );
+
+  extendArrayProto('withPrevAndCurrent', function (mapFn: any) {
+    /* eslint-disable babel/no-invalid-this */
+    // @ts-ignore afesfa
+    return withPrevAndCurrent(this, mapFn);
+    /* eslint-enable */
+  });
+
+  const events = eventList
+    .map(fillMissingDefaultsFor({ handlerTag, handlerType }))
+    .withPrevAndCurrent(fillMissingActiveStateFields)
+    .withPrevAndCurrent(fillOldStateChanges)
+    .withPrevAndCurrent(validateStateTransitions)
```

Comment:
Those functions are badly typed, we need to revise the internal handlers typing.